src/OnChain.ts

```diff
@@ -200,33 +211,58 @@ export const OnChainMixin = (superclass) =>
             throw new LoggedError(error)
           }
 
-          // case where there is not enough money available within lnd on-chain wallet
-          if (onChainBalance < amount + estimatedFee) {
-            // TODO: add a page to initiate the rebalancing quickly
-            throw new RebalanceNeededError(undefined, {
-              logger: onchainLogger,
-              onChainBalance,
-              amount,
-              estimatedFee,
-              sendTo,
-              success: false,
-            })
-          }
+          if (!sendAll) {
+            amountToSend = amount
+
+            // case where there is not enough money available within lnd on-chain wallet
+            if (onChainBalance < amount + estimatedFee) {
+              // TODO: add a page to initiate the rebalancing quickly
+              throw new RebalanceNeededError(undefined, {
+                logger: onchainLogger,
+                onChainBalance,
+                amountToSend,
```

Comment:
Is the rename from 'amount' to 'amountToSend' ok for the error metadata?

Reply:
Any reason for the rename here?

Reply:
Hello Kartik! That was originally implemented that way, but then I decided to add the named parameter so that no change would be needed. https://github.com/GaloyMoney/galoy/pull/274/commits/eb8e7ac257be22672ba52632e39b477044c7eaff